autogen/gradio_gui/README.md

```diff
@@ -0,0 +1,104 @@
+# AutoGen's Gradio GUI
+
+This is a GUI for AutoGen. It is written in Python and uses the [Gradio](https://gradio.app/) library.
+
+## ⭐Installation
+
+1. install AutoGen:
+
+```bash
+python -m pip install pyautogen
+```
+
+## ⭐Run
+
+1. create `OAI_CONFIG_LIST`, write your OpenAI API key or azure API key. For example:
+
+```
+[
+    {
+        "model": "gpt-3.5-turbo-16k",
+        "api_key": "----------------------------------",
+        "api_base": "https://--------.openai.azure.com",
+        "api_type": "azure",
+        "api_version": "2023-07-01-preview",
+        "deployment_id": "------"
+    }
+]
+
+```
+
+2. run following commands launch GUI:
+
+```bash
+export OAI_CONFIG_LIST='/path/to/OAI_CONFIG_LIST'
+export AUTOGEN_USE_DOCKER='False'
+export WEB_PORT=12345
+# for more environment variables options, please refer to the void terminal project
+
+python -m autogen.launch_gui
+```
+
+> Note:
+> When you run `launch_gui`, the program will automatically install additional dependencies with `pip install --user ...` if they are not installed.
+
+## ⭐Customize
+
+
+1. Create a python file `test.py` with templete below
+2. Edit and add your own multiagent chat
+3. Run it with `python test.py`
+
+```python
+
+from autogen.gradio_gui import init_config
+from autogen.gradio_gui.utils.general import AutoGenGroupChat
+from autogen.gradio_gui.plugin import autogen_terminal
+from autogen.gradio_gui.gradio_service import main
+import os
+llm_config = init_config()
+
+# <-------------------  define autogen agents (group chat)  ------------------->
+class AutoGenGroupChat(AutoGenGroupChat):
```

Comment:
It would be nice to explain more details about the design and functionalities of the functions to be overridden.

Reply:
added in samples/apps/gradio_gui/general.py